crates/bifrost/src/watchdog.rs

```diff
@@ -125,11 +186,43 @@ impl Watchdog {
             Some(cmd) = self.inbound.recv() => {
                 self.handle_command(cmd)
             }
+            Some((log_id, trim_request)) = self.trims_recv.recv() => {
+                self.store_trim_request(log_id, trim_request);
+            }
+            Some(Ok(trim_result)) = conditional_await(self.in_flight_trim.as_mut()) => {
+                self.in_flight_trim = None;
+
+                match trim_result {
+                    Ok(()) => {
+                        // successful trim
+                    },
+                    Err(failed_trims) => {
+                        // trim failed, merge them back into the map for the next attempt
+                        for (log_id, trim_request) in failed_trims {
+                            self.store_trim_request(log_id, trim_request)
+                        }
```

Comment:
Should we track attempts and give up on entries at some point? I can't imagine why but there could be a scenario where one failing key loops around forever, blocking all progress. `</paranoia>`

Reply:
i believe our success/failure is simply about a metadata operation, so either no mutations are done or they are all done. however this will change if we move to doing the trim point lookup after this point - in that case technically looking it up could fail for some logs and not others. but i think i can handle that